spacy/kb/candidate.py

```diff
@@ -0,0 +1,109 @@
+import abc
+from typing import List, Union, Callable
+
+
+class BaseCandidate(abc.ABC):
+    """A `BaseCandidate` object refers to a textual mention (`alias`) that may or may not be resolved
+    to a specific `entity_id` from a Knowledge Base. This will be used as input for the entity_id linking
+    algorithm which will disambiguate the various candidates to the correct one.
+    Each candidate (alias, entity_id) pair is assigned a certain prior probability.
+
+    DOCS: https://spacy.io/api/kb/#candidate-init
+    """
+
+    def __init__(
+        self, mention: str, entity_id: Union[int, str], entity_vector: List[float]
+    ):
+        """Create new instance of `Candidate`. Note: has to be a sub-class, otherwise error will be raised.
+        mention (str): Mention text for this candidate.
+        entity_id (Union[int, str]): Unique entity ID.
+        entity_vector (List[float]): Entity embedding.
+        """
```

Comment:
TODO: update doc string

Reply:
I didn't see the commit addressing this?

Reply:
To clarify: the mention of "Candidate" and it having to be a sub-class.

Reply:
Ah, my push timed out when my connection dropped :smiling_face_with_tear:

Reply:
It's this one: https://github.com/explosion/spaCy/pull/12341/commits/3da0712582e272bd757cfe41a6a477c9d875bd9e